src/test/playwright/support/pageobjects/exercises/programming/RepositoryPage.ts

```diff
@@ -13,16 +14,20 @@ export class RepositoryPage {
     }
 
     async checkCommitHistory(commits: ExerciseCommit[]) {
+        // Reverse commit history as the latest commit in commit history table is at the bottom
+        commits = commits.reverse();
         const commitHistory = this.page.locator('.card-body', { hasText: 'Commit History' });
 
         if (commits) {
             const commitCount = commits.length;
-            for (let index = 0; index < commitCount; index++) {
+            for (let index = commitCount - 1; index > 0; index--) {
```

Comment:
What is the advantage of reversing the commits and then iterating over the commits from top to bottom?

I think that the previous version was easier to understand. Am I missing a change in semantics or an optimization here?

Reply:
Here I reverse to have easier-looking indexing - `commits[index]` instead of `commits[commitCount - index - 1]`, because ordering is opposite in commit history web elements. It's a slightly added complexity, do you think I should not do that?

I traverse starting from the bottom because the first build to finish is that one, and while we're asserting the first commit, the next one is building. In practice no visible time difference, but I think it still makes sense to start with the chronologically first build.

Reply:
For me the current approach is harder to understand - however, this might be personal preference 🤷 

For me the following approach would be easiest to read and understand
```
readonly LATEST_COMMIT_INDEX = 0;
for (let index = LATEST_COMMIT_INDEX; index < commitCount; index++) {
```

Reply:
Unfortunately didn't understand what a zero constant would be helpful for. Instead used initial commit index as a constant reference for calculating table indices and got rid of reversing commits. Do you think it's readable now?